src/fciqmc.jl

```diff
@@ -0,0 +1,191 @@
+"""
+    FCIQMC(; kwargs...) <: PMCAlgorithm
+
+Algorithm for the full configuration interaction quantum Monte Carlo (FCIQMC) method.
+The default algorithm for [`ProjectorMonteCarloProblem`](@ref).
+
+# Keyword arguments and defaults:
+- `shift_strategy = DoubleLogUpdate(; targetwalkers = 1_000, ζ = 0.08,
+    ξ = ζ^2/4)`: How to update the `shift`.
+- `time_step_strategy = ConstantTimeStep()`: Adjust time step or not.
+
+See also [`ProjectorMonteCarloProblem`](@ref), [`ShiftStrategy`](@ref),
+[`TimeStepStrategy`](@ref), [`DoubleLogUpdate`](@ref), [`ConstantTimeStep`](@ref).
+"""
+Base.@kwdef struct FCIQMC{SS<:ShiftStrategy,TS<:TimeStepStrategy} <: PMCAlgorithm
+    shift_strategy::SS = DoubleLogUpdate()
+    time_step_strategy::TS = ConstantTimeStep()
+end
+function Base.show(io::IO, a::FCIQMC)
+    print(io, "FCIQMC($(a.shift_strategy), $(a.time_step_strategy))")
+end
+
+"""
+    set_up_initial_shift_parameters(algorithm::FCIQMC, hamiltonian,
+    starting_vectors, shift, time_step, initial_shift_parameters
+)
```

Comment:
This doesn't display nicely. How about
```suggestion
    set_up_initial_shift_parameters(
        ::FCIQMC, hamiltonian, starting_vectors, shift, time_step, initial_shift_parameters
    )
```